src/FsAutoComplete/LspHelpers.fs

```diff
@@ -673,7 +673,8 @@ type FSharpConfigDto =
     LineLens: LineLensConfig option
     UseSdkScripts: bool option
     DotNetRoot: string option
-    FSIExtraParameters: string array option
+    FSIExtraInteractiveParameters: string array option
+    FSIExtraSharedParameters: string array option
```

Comment:
When we take this update in Ionide we'll need to use the deprecation features of the config system there to message to users that this setting has changed, but that's a reasonable thing to change IMO.

Reply:
Should we do similar here in FSAC?

Reply:
I'll add the old field back again. I will leave the changes to where the options are consumed, so that all consuming code is referencing the new `FSIExtraSharedParameters`. And then I think the logic below should set us up well for success now and in the future:

- `FSIExtraParameters` is set explicitly *and* the new params are not set -> set `FSIExtraSharedParameters` to the configured value of `FSIExtraParameters`.
    - Reasoning: they are using the old field, so expecting unchanged behavior. The new code consumes only `FSIExtraSharedParameters`. By setting the new option in this way, everything the client has passed to us is used, mimicking the existing behavior.
- `FSIExtraParameters` not set explicitly -> use whatever is in the new parameters.
    - Reasoning: if they are not setting the old parameter, then it doesn't matter. We can unconditionally take the new ones (or defaults if they are blank)
- `FSIExtraParameters` is set explicitly *and* the new params are set -> unsupported. We take the new params only and ignore `FSIExtraParameters`. Maybe log a warning or `raise` in this case?
    - Reasoning: if someone is using the new parameters, they should be opting in to new behavior. If they're using all three, there is no reasonable thing to do with the old parameter. Given that this depends on them opting in to new parameters, I'd argue to `raise` and tell them to pick one.

Following this logic, when we eventually remove `FSIExtraParameters`, all we need to do is remove the bit that conditionally copies its value to `FSIExtraSharedParameters`.

Thoughts? Specifically:
1. Do you like the conditional logic I laid out above?
2. What to do if the client is sending config options for the combination of both the old and the new parameters? I like `raise`ing on this.

Reply:
I generally like this with the exception that we shouldn't `raise` because we only consume these changes in two RPC calls:
* initialize
* workspace/didChangeConfiguration

`raise` in either of these contexts will either leave the LSP in an unstable state or prevent the configuration change from 'taking'. Instead, we should send a message back to the client via the `lspClient.WindowShowMessage` API, like we do for other warnings/notifications. You can see an example of this [here](https://github.com/ionide/FsAutoComplete/blob/08b871146e6c9d5368c1515dba1dd4f14c9b6548/src/FsAutoComplete/LspServers/AdaptiveFSharpLspServer.fs#L151-L154). This should communicate the situation to users while not impacting the stability of the LSP.

Reply:
Sounds good. I think sticking to that requirement actually simplifies the logic a bit:

- `FSIExtraParameters` is sent by client and no new params -> silently use this as `FSIExtraSharedParameters`, which mimics current behavior. when we want, in the future, we can also start messaging about deprecation for this.
- `FSIExtraParameters` is sent by client and either/both new params -> use this as `FSIExtraSharedParameters` and send a message to that effect, warning not to mix old and new
- no `FSIExtraParameters` -> silent, use whatever is passed (or not) by client for new parameters